azurerm/helpers/azure/key_vault.go

```diff
@@ -0,0 +1,121 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/Azure/azure-sdk-for-go/services/keyvault/mgmt/2018-02-14/keyvault"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func GetKeyVaultBaseUrlFromID(ctx context.Context, client keyvault.VaultsClient, keyVaultId string) (string, error) {
+
+	if keyVaultId == "" {
+		return "", fmt.Errorf("keyVaultId is empty")
+	}
+
+	id, err := ParseAzureResourceID(keyVaultId)
+	if err != nil {
+		return "", err
+	}
+	resourceGroup := id.ResourceGroup
+
+	vaultName, ok := id.Path["vaults"]
+	if !ok {
+		return "", fmt.Errorf("resource id does not contain `valuts`: %q", keyVaultId)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, vaultName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return "", fmt.Errorf("Error unable to find KeyVault %q (Resource Group %q): %+v", vaultName, resourceGroup, err)
+		}
+		return "", fmt.Errorf("Error making Read request on KeyVault %q (Resource Group %q): %+v", vaultName, resourceGroup, err)
+	}
+
+	if resp.Properties == nil || resp.Properties.VaultURI == nil {
+		return "", fmt.Errorf("vault (%s) response properties or VaultURI is nil", keyVaultId)
+	}
+
+	return *resp.Properties.VaultURI, nil
+}
+
+func GetKeyVaultIDFromBaseUrl(ctx context.Context, client keyvault.VaultsClient, keyVaultUrl string) (string, error) {
+
+	list, err := client.ListComplete(ctx, utils.Int32(1))
```

Comment:
this'll only return a single keyvault - from the SDK docs:
> // top - maximum number of results to return.

in other resources we set this to 1000 - I'd suggest it's worth doing the same here

```suggestion
	list, err := client.ListComplete(ctx, utils.Int32(1000))
```